client/src/index.jsx

```diff
@@ -1,13 +1,202 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import {Header} from './header.jsx';
+import {AnimalSet} from './animalSet.jsx';
+import {NewSighting} from './addSighting.jsx';
 import '../style.sass';
 
-const element = <h1 className='test'>Hello, world</h1>;
+class App extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			sightings: [sight, sight1, sight2],
+			showing: "sight"
+		
+		};
+		this.viewHomepage = this.viewHomepage.bind(this);
+		this.addSighting = this.addSighting.bind(this);
+		this.viewAnimals = this.viewAnimals.bind(this);
+		this.createNewSighting = this.createNewSighting.bind(this);
 
-console.log("Running!");
+	}
+
+
+	renderSightings(){
+		const {sightings} = this.state;
+		if (!sightings | sightings.length < 0) return null;
+		const SATList = []; //Create list of SightingAsText elements
+		for (let i = 0; i < sightings.length; i+=1) {
+			SATList.push(
+				<SightingAsText user={person} sighting={sightings[i]} />
+			);
+		}
+		return SATList;
+	};
+
+	viewHomepage() {
+		console.log("Clicked Home");
+		this.setState({
+			showing: "sight"
+		});
+	}
+
+	addSighting() {
+		console.log("Clicked Sighting");
+		this.setState(state => ({
+			//sightings: [...state.sightings, sight],
+			showing: "newSight"
+		}));
+	}
+
+	viewAnimals() {
+		console.log("Clicked Animals");
+		this.setState({
+			showing: "animals"
+		});
+	}
+
+
+	showAnimals() {
+		console.log("Showing animals");
+		const {sightings} = this.state;
+		if (!sightings | sightings.length < 0) return null;
+		const animals = {};
+
+		//Create "set" of animals
+		for (let i = 0; i < sightings.length; i+=1) {
+			let sight = sightings[i];
+			if (sight.animal in animals) {
```

Comment:
If you're checking if something exists in an object, you can also just try accessing it.

`if ( animals[sight.animal] ) ` 

I didnt even know this worked tbh